search-index/test/test_text2embeddings.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+import text2embeddings
+
+
+def test_get_text_from_merged_block():
+    # Case 1: Single superscript. c.f cclw-10007. Superscript has been removed.
+    input_1 = {
+        "text": [
+            "The central indicator for measuring the depletion of natural resources at the macroeconomic level is the environmentally adjusted GDP: GDP \u2013 GDPE . It is the adjustment of GDP by the depletion (depreciation) of natural capital, which is the consumption of environmental products. Consumption of environmental products means quantitative reduction of minerals, bio-resources and the observed reduction in ecosystem services.  "
+        ],
+        "text_block_id": "p6_b91_merged",
+        "coords": [
+            [85.08000183105469, 207.0596923828125],
+            [555.9367065429688, 207.0596923828125],
+            [85.08000183105469, 278.4237060546875],
+            [555.9367065429688, 278.4237060546875],
+        ],
+        "type": "merged_text_block",
+        "path": ["Document", "P[40]"],
+        "custom_attributes": {
+            "styleSpans": [{"style": "superscript", "start_idx": 144, "end_idx": 144}]
```

Comment:
Only just noticed this but the start_idx and end_idx behaviour is a bit un-pythonic here - you'd expect the representation of only index 144 to be (144,145). Happy to leave it like this as it looks like other code might depend on it, as long as a note about this behaviour is in the relevant docstring.